src/sentry/static/sentry/app/components/group/externalIssueActions.jsx

```diff
@@ -151,114 +153,189 @@ class ExternalIssueActions extends AsyncComponent {
-  static propTypes = {
-    group: PropTypes.object.isRequired,
-    sentryAppComponent: PropTypes.object.isRequired,
-    sentryAppInstallation: PropTypes.object,
-    externalIssue: PropTypes.object,
-  };
+export const SentryAppExternalIssueActions = withApi(
+  class SentryAppExternalIssueActions extends React.Component {
+    static propTypes = {
+      api: PropTypes.object.isRequired,
+      group: PropTypes.object.isRequired,
+      sentryAppComponent: PropTypes.object.isRequired,
+      sentryAppInstallation: PropTypes.object,
+      externalIssue: PropTypes.object,
+    };
 
-  constructor(props) {
-    super(props);
+    constructor(props) {
+      super(props);
 
-    this.state = {
-      action: 'create',
-      showModal: false,
+      this.state = {
+        action: 'create',
+        externalIssue: props.externalIssue,
+        showModal: false,
+      };
+    }
+
+    componentWillUpdate(prevProps) {
+      if (this.props.externalIssue !== prevProps.externalIssue) {
+        this.setState({externalIssue: this.props.externalIssue});
+      }
+    }
+
+    showModal = () => {
+      // Only show the modal when we don't have a linked issue
+      !this.state.externalIssue && this.setState({showModal: true});
     };
-  }
 
-  showModal = () => {
-    // Only show the modal when we don't have a linked issue
-    !this.props.externalIssue && this.setState({showModal: true});
-  };
+    hideModal = () => {
+      this.setState({showModal: false});
+    };
 
-  hideModal = () => {
-    this.setState({showModal: false});
-  };
+    showLink = () => {
+      this.setState({action: 'link'});
+    };
 
-  showLink = () => {
-    this.setState({action: 'link'});
-  };
+    showCreate = () => {
+      this.setState({action: 'create'});
+    };
 
-  showCreate = () => {
-    this.setState({action: 'create'});
-  };
+    deleteIssue = () => {
+      const {api, group} = this.props;
+      const {externalIssue} = this.state;
+      const url = `/issues/${group.id}/external-issues/${externalIssue.id}/`;
 
-  iconExists() {
-    try {
-      require(`../../icons/${this.props.sentryAppComponent.sentryApp.slug}.svg`);
-      return true;
-    } catch (err) {
-      return false;
+      api.request(url, {
+        method: 'DELETE',
+        success: data => {
+          this.setState({externalIssue: null});
+          addSuccessMessage(t('Successfully unlinked issue.'));
+        },
+        error: error => {
+          addErrorMessage(t('Unable to unlink issue.'));
+        },
+      });
+    };
+
+    onAddRemoveClick = () => {
+      const {externalIssue} = this.state;
+
+      if (!externalIssue) {
+        this.showModal();
+      } else {
+        this.deleteIssue();
+      }
+    };
+
+    onSubmitSuccess = externalIssue => {
+      this.setState({externalIssue});
+      this.hideModal();
+    };
+
+    iconExists() {
+      try {
+        require(`../../icons/${this.props.sentryAppComponent.sentryApp.slug}.svg`);
```

Comment:
Will this do a network call on each render?

Reply:
I'm not sure, to be totally honest. I don't know how node/React does things here.

Reply:
This is more of a webpack thing, I don't remember how `require` works, but I think it does make a network call.

Reply:
fwiw, I don't see an explicit request in my browser for this. 🤷‍♂️

Reply:
It's possible then that webpack just includes everything in that directory in the main bundle